AppConnectSwift/CaptureImage.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  CaptureImage.swift
+//  AppConnectSwift
+//
+//  Created by Suman Sucharit Das on 4/1/16.
+//  Copyright © 2016 Medidata Solutions. All rights reserved.
+//
+
+import UIKit
+
+class CaptureImage: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    
+    @IBOutlet weak var imageView: UIImageView!
+    var imagePicker = UIImagePickerController()
+    var image = UIImage()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        photoUpload(true)
+    }
+    
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        self.navigationItem.title = "Capture Image"
+    }
+    
+    // MARK: Image picker delegate functions
+    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : AnyObject]) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+        print(info)
+        let image = info[UIImagePickerControllerOriginalImage] as? UIImage
+        imageView.image = image
+        self.image = image!
+    }
+    
+    func imagePickerControllerDidCancel(picker: UIImagePickerController) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+    }
+    
+    // MARK: View Controller Toolbar Items
+    @IBAction func searchTapped(sender: AnyObject) {
+        photoUpload(false)
+    }
+    
+    @IBAction func cameraTapped(sender: AnyObject) {
+        photoUpload(true)
+    }
+    
+    @IBAction func saveTapped(sender: AnyObject) {
+        if(image.CGImage != nil){
+            let data = compressFile() as? NSData
+            var subject: MDSubject!
+            subject.collectData(data, withMetadata: "Random String", completion: { (dataEnvelop: MDSubjectDataEnvelope!, err: NSError!) -> Void in
+                if(err != nil){
+                    print(err?.description);
+                }
+                else{
+                    print("Data Saved");
+                }
+            })
+            // Empty the view once data successfully saved
+            imageView.image = nil
+        }
+        else{
+            postAlert("No image", message: "Image selected has no path")
+        }
+    }
+    
+    func photoUpload(fromCamera: Bool) {
```

Comment:
I think there could be a better name for this - maybe `takeOrSelectPicture` or something along those lines? It's unclear what `photoUpload` means.